### PR TITLE
Delete Unused Bits of StateService

### PR DESCRIPTION
**Pull request by justindbaur**
## 📔 Objective

Password generator related data has now fully removed all references to their legacy `StateService` methods/data. This means we can do a fair amount of cleanup. We can also get rid of all derived implementations of `StateService` except for web. 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/core/state/global-state.ts [hunk 1]
```diff
@@ -1,5 +0,0 @@
-import { GlobalState as BaseGlobalState } from "@bitwarden/common/platform/models/domain/global-state";
-
-export class GlobalState extends BaseGlobalState {
-  rememberEmail = true;
```
Comment (justindbaur) [inline apps/web/src/app/core/state/global-state.ts:4]: @bitwarden/team-auth-dev I noticed this while deleting things that web doesn't default to rememberEmail being true when it was migrated outside of state service. Not sure if that was intentional or not.

**Review by dani-garcia — APPROVED** [review]

**Review by audreyality — DISMISSED** [review]
👍🏻 Thank you for cleaning up some of the generator state! My next task is cleaning up the MV2 code, and that makes it simpler! 😄

**Review by jprusik — DISMISSED** [review]
Approved for autofill concerns

**Review by dani-garcia — APPROVED** [review]

**Review by audreyality — APPROVED** [review]

File: apps/web/src/app/core/state/global-state.ts (hunk 1 above)
Comment (jlf0dev) [reply apps/web/src/app/core/state/global-state.ts:4]: Thanks for the heads up!

**Review by jprusik — APPROVED** [review]

**Pull request merged**